### DISCUSSION TITLE
Replacing an Entry Type field with another field while maintaining the content

### DISCUSSION THREAD

**Discussion by svondervoort** [craftcms/cms#15814] [Q&A] [answered]
We have a "Plain Text" field which we would like to replace with a CKEditor field.

Now I know it is possible to just change the field configuration to a CKEditor, but we use the same field in different Entry Types where it need to stay as a "Plain Text" field.

Is there a way to "replace" the field while keeping the content or a way to migrate the data?

**Comment by brandonkelly** [comment, marked as answer]
You can create a new CKEditor field, attach it to the entry type(s) you wish to use it on, and then populate it with values from the original Plain Text field using the `resave/entries` command:

```sh
php craft resave/entries --type myEntryType --set myCkeditorField --to myPlainTextField
```

Or if the Plain Text field isn’t currently storing HTML, you can wrap its values in a `<p>` tag:

```sh
php craft resave/entries --type myEntryType --set myCkeditorField \
  --to "fn(\$entry) => sprintf('<p>%s</p>', (string)\$entry->myPlainTextField)"
```

Once you’ve done that and verified the CKEditor fields look right (on all environments), you can remove the Plain Text field from the entry type(s) it’s no longer needed on.

**Reply by svondervoort** [reply]
Thanks. The `resave/entries` command has more up it's sleeve than I thought :)

**Reply by brandonkelly** [reply]
It’s a pretty handy command :)
